### PR TITLE
added tests for primitive types, dictionary type, and enumerable type for convertto-xml

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
Based on https://codecov.io/gh/PowerShell/PowerShell/src/master/src/Microsoft.PowerShell.Commands.Utility/commands/utility/CustomSerialization.cs

Since I'm going to be on vacation, I'm submitting PR now rather than when I finish going through the whole file

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Xml.Tests.ps1 [hunk 1]
```diff
@@ -67,6 +67,43 @@
...
+    }
+
+    It "Serialize primitive type" {
+        [int] $i = 1
+        $x = $i | ConvertTo-Xml
+        $x.Objects.Object.Type | Should BeExactly $i.GetType().ToString()
+        $x.Objects.Object."#text" | Should BeExactly $i
+    }
+
+    It "Serialize dictionary type" {
```
Comment (anmenaga) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Xml.Tests.ps1]: If you're using terminology from the source code, then I would call this test
"Serialize ContainerType.Dictionary type"

File: test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Xml.Tests.ps1 [hunk 2]
```diff
@@ -67,6 +67,43 @@
...
+        $a = @{foo="bar"}
+        $x = $a | ConvertTo-Xml
+        $x.Objects.Object.Type | Should BeExactly $a.GetType().ToString()
+        $x.Objects.Object.Property[0].Name | Should BeExactly "Key"
+        $x.Objects.Object.Property[0]."#text" | Should BeExactly "foo"
+        $x.Objects.Object.Property[1].Name | Should BeExactly "Value"
+        $x.Objects.Object.Property[1]."#text" | Should BeExactly "bar"
+    }
+
+    It "Serialize enumerable type" {
```
Comment (anmenaga) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Xml.Tests.ps1]: Like above; maybe rename the test to "Serialize ContainerType.Enumerable type"

File: test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Xml.Tests.ps1 [hunk 3]
```diff
@@ -67,6 +67,43 @@
...
+            [string] $name;
+        }
+
+        $fruit1 = [fruit]::new()
+        $fruit1.name = "apple"
+        $fruit2 = [fruit]::new()
+        $fruit2.name = "banana"
+        $x = $fruit1,$fruit2 | ConvertTo-Xml
+        $x.Objects.Object.Count | Should BeExactly 2
+        $x.Objects.Object[0].Type = "fruit"
```
Comment (anmenaga) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Xml.Tests.ps1]: For this line and below:
Are these assignments ("=") meant to be verifications ( | Should ) ?

**Review by anmenaga — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Xml.Tests.ps1 (hunk 1 above)
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Xml.Tests.ps1]: ok

File: test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Xml.Tests.ps1 (hunk 2 above)
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Xml.Tests.ps1]: ok

File: test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Xml.Tests.ps1 (hunk 3 above)
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Xml.Tests.ps1]: yeah, that's a mistake

**Review by adityapatwardhan — APPROVED** [review]

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]
Please fix CI errors.

**Comment by adityapatwardhan** [comment]
CI failing with:

Expected string length 5 but was 27. Strings differ at index 0.
Expected: {fruit}
But was:  {<2f125142>.<bbb5d5d6>.fruit}
-----------^

**Comment by SteveL-MSFT** [comment]
@anmenaga @adityapatwardhan feedback addressed

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@anmenaga Can you have a look?

**Review by anmenaga — APPROVED** [review]

**Pull request merged**